### ISSUE TITLE
Crash upon Nether Generation

### ISSUE THREAD

**Issue by Mey-Ha-Zah** [nekomaster1000/Infernal-Expansion#367]
### Describe the Bug

Another Generation crash bug, here's the crash log. https://pastebin.com/ctB4w1vG

### Reproduction Steps

Just entered the Nether at a Portal I made in a new world.

### Expected Result

Enter the Nether

### Operating System

Windows 10

### Mod Version

2.5.1

### Minecraft Version

1.19

### Forge Version

41.0.63

### Other Mods

Appleskin, Bookshelf, Clumps, Comforts (forge edition), Cosmetic Armor Reworked, Enchantment Descriptions, Fast Leaf Decay, Goblin Traders, Gravestone Mod, Iron Chests, Mining Master, Moonlight Lib, More Villagers, Mouse Tweaks, Sophisticated Backpacks, Sophisticated Core, Supplementaries

### Additional Context

Found a cool generated portal frame, started building a base, entered the nether for some resource havesting and this happened.

**Comment by Mey-Ha-Zah** [comment]
I can even provide the seed and the portal frame location which I used which caused the crash if need be.

**Comment by Dariensg** [comment]
Sorry this issue came back up! I wasn't able to replicate it before, but found what seemed to be causing the issue anyway. I'm looking into it again to see if I can find another instance of the error that I missed previously. I'll keep you updated!

**Comment by Dariensg** [comment]
Could you do me a huge favor and replicate the issue with [this](https://www.curseforge.com/minecraft/mc-mods/cyanide-forge) mod installed? That should give more information if the crash still occurs. If it doesn't occur, it's possible that the crash is caused by [this](https://bugs.mojang.com/browse/MC-252369) which I believe Cyanide should fix as of version 2.1.1. Thanks in advance!

**Comment by baooduy** [comment]
Install Cyanide fixed the problem for me

**Comment by Mey-Ha-Zah** [comment]
Well I did and found out something interesting. https://pastebin.com/7YrX4SPF

I think this will best explain it.

**Comment by Dariensg** [comment]
Interesting! Well, I’ll have to figure that out, but it might actually be a Mining Master issue. I’ll have to figure it out, so I’ll leave this issue open here for now until I determine which mod is actually causing the issue. Thanks for reporting this!

**Comment by Glukose69** [comment]
I've done some personal bug testing and I've found that both terrablender and moonlight cause crashes with nether generation

**Comment by Dariensg** [comment]
Do you mean in conjunction with Infernal Expansion or on their own? There is an existing Vanilla bug that could cause these exact crashes. I haven't yet had a chance to look at Mining Master to see what could be causing this, but the crash does seem to be a strange issue that shouldn't be occurring (especially since the Cyanide log seems to say that the feature should be before itself if I'm reading it correctly). It would be interesting to know if TerraBlender is having similar issues as well.

**Comment by Dariensg** [comment]
The issue has been identified and there will hopefully be a fix soon. There's a possibility Forge themselves may rework something that would fix this issue, but, if they decide not to, we will fix this on our end. Thank you for your patience!

**Comment by Mey-Ha-Zah** [comment]
When the fix is made, is it possible to update the world's data pack so I can keep the world?

**Comment by Dariensg** [comment]
I'm not 100% sure, but I'd assume there is a way, yes. I've never attempted it, but I think there should be some way to do it

**Comment by Shadowex3** [comment]
I might have run into this as well, the pastebins are expired so I can't tell. I had a feature cycle crash on attempting to enter the nether the first time.  

Edit: False alarm, it was Geodes doing it for me.

```
Cycle:
At step 6
Feature '[Unknown minecraft:worldgen/placed_feature]'
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:crystalline_chasm' at index 18, 19 and 5 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'minecraft:nether_wastes' at index 3, 4 and 3 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 2, 3 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 3, 4 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 4, 5 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 5, 6 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 6, 7 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 7, 8 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 8, 9 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 9, 10 and 8 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 10, 11 and 8 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 11, 12 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 12, 13 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 13, 14 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 14, 15 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 15, 16 and 10 others)
  must be before '[Unknown minecraft:worldgen/placed_feature]' (defined in 'biomesoplenty:undergrowth' at index 16, 17 and 1 others)
```

**Comment by Dariensg** [comment]
Fixed by 2395c299a77c86ddcd7c33ea7f4bb437a7038115. Sorry didn't realize this was still open!
